clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -225,22 +226,27 @@ public static Properties consumerConfig(final String bootstrapServers, Class key
     }
 
     /**
-     *  uses default value of 30 seconds for timeout
+     *  uses default value of 15 seconds for timeout, and default message for failure
      */
     public static void waitForCondition(TestCondition testCondition) throws InterruptedException {
-        waitForCondition(testCondition, 30000);
+        waitForCondition(testCondition, 15000, "Expected condition not met by timeout");
     }
 
     /**
-     *  Used to wait for specific conditions/state to be me during a test
-     *  this is meant to be a replacement for using Thread.sleep
+     *  Used to wait for specific conditions/state to be met during a test.
+     *  It's best to avoid using Thread.sleep in tests
+     *  as transient timing failures can occur and could make the test suit take longer to run.
```

Comment:
I would say something like:

```java
/**
 * Wait for condition to be met for at most {@code maxWaitMs} and throw assertion failure otherwise.
 * This should be used instead of {@code Thread.sleep} whenever possible as it allows a longer timeout to be used
 * without unnecessarily increasing test time (as the condition is checked frequently). The longer timeout is needed to
 * avoid transient failures due to slow or overloaded machines. 
 */
```